### PR TITLE
threaded-dest: make persist_name local

### PR DESCRIPTION
**Pull request by alltilla**
Before this commit, the worker's queue's `persist_name` was stored in a static string, and it was accessed and changed by multiple threads in the same time.

The static string was there to avoid having to care about malloc/free, it is not performance critical, so it can be changed to local.

Follow up for the comments at #2512.

Signed-off-by: Attila Szakacs <attila.szakacs@balabit.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/logthrdestdrv.c [hunk 1]
```diff
@@ -76,24 +76,21 @@ log_threaded_dest_worker_rewind_messages(LogThreadedDestWorker *self, gint batch
   self->batch_size -= batch_size;
 }
 
-static const gchar *
-_format_queue_persist_name(LogThreadedDestWorker *self)
+static void
+_format_queue_persist_name(LogThreadedDestWorker *self, gchar *persist_name, size_t len)
 {
-  static gchar persist_name[1024];
   LogPipe *owner = &self->owner->super.super.super;
```
Comment (bazsi) [inline lib/logthrdestdrv.c]: I would probably convert this (and other similar call sites) to use g_strdup_printf() and ensure the result is freed by the caller.

That way we don't have to care about 1024 character limits.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/logthrdestdrv.c [hunk 2]
```diff
@@ -76,24 +76,21 @@ log_threaded_dest_worker_rewind_messages(LogThreadedDestWorker *self, gint batch
   self->batch_size -= batch_size;
 }
 
-static const gchar *
+static gchar *
```
Comment (furiel) [inline lib/logthrdestdrv.c]: Why did you need to remove const? `g_strdup_printf` indeed returns `gchar *`, but I think that is converted automatically without warnings to `const gchar *`.

**Comment by kira-syslogng** [comment]
Build SUCCESS
